src/tile.js

```diff
@@ -173,18 +180,19 @@ export default class Tile {
                         }
                     }
 
-                    tile.debug.features++;
+                    source.debug.features++;
                 }
+
             }
-        }
 
-        // Finalize array buffer for each render style
-        tile.vertex_data = {};
-        for (var m in vertex_data) {
-            tile.vertex_data[m] = vertex_data[m].end().buffer;
-        }
+            // Finalize array buffer for each render style
```

Comment:
This should be outside this loop, after all sources are complete. Having it here may cause undefined results.